tree/dataframe/inc/ROOT/RDF/RResultMap.hxx

```diff
@@ -91,6 +102,28 @@ public:
 
 } // namespace Experimental
 } // namespace RDF
+
+namespace Detail {
+namespace RDF {
+template <typename T>
+std::unique_ptr<RMergeableVariations<T>> GetMergeableValue(ROOT::RDF::Experimental::RResultMap<T> &rmap)
+{
+   if (!rmap.fAction->HasRun())
+      rmap.fLoopManager->Run(); // Prevents from using `const` specifier in parameter
+
+   auto type_erased_variations = std::unique_ptr<RTypeErasedMergeables>{
+      static_cast<RTypeErasedMergeables *>(rmap.fAction->GetMergeableValue().release())};
+
+   std::vector<std::unique_ptr<RMergeableValue<T>>> values;
+   values.reserve(type_erased_variations->GetNValues());
+   for (auto &&mergeablebase : type_erased_variations->GetValues()) {
+      values.emplace_back(static_cast<RMergeableValue<T> *>(mergeablebase.release()));
+   }
+
+   return std::make_unique<RMergeableVariations<T>>(type_erased_variations->GetKeys(), std::move(values));
```

Comment:
iiuc this basically performs a conversion from `RTypeErasedMergeables` to `RMergeableVariations<T>`. I am not sure this is needed (see main review comment), but if it is, this logic should be shoved into a conversion operator so that here it can be immediately clear what's going on, i.e. these lines should just become:

```cpp
return std::make_unique<RMergeableVariations<T>(typeErasedVariations);
```